stock_picking_partner_info/views/stock_picking_views.xml

```diff
@@ -12,6 +12,12 @@
                     name="attrs"
                 >{'invisible': [('picking_type_code', '!=', 'outgoing')]}</attribute>
             </xpath>
+            <xpath expr="//field[@name='partner_id']" position="after">
+                <field
+                    name="partner_id"
+                    attrs="{'invisible': [('picking_type_code', '=', 'outgoing')]}"
+                />
+            </xpath>
```

Comment:
```.xml
            <xpath expr="//field[@name='partner_id']" position="attributes">
                <attribute
                    name="attrs"
                >{'invisible': [('picking_type_code', '=', 'outgoing')]}</attribute>
            </xpath>
            <xpath expr="//field[@name='partner_id']" position="after">
                <field
                    name="partner_id"
                    context="{'show_address': 1}"
                    options="{'always_reload': 1}"
                    attrs="{'invisible': [('picking_type_code', '!=', 'outgoing')]}"
                />
            </xpath>
```
Slightly better this way?

Reply:
I got it. I agree it is slightly easy to read.

Reply:
And this is a kind notation for the reader as there is another field of phone after this.

Reply:
I sort these fields as below to adjust layout.
1. patrner_id w/ "invisible when picking_type_code = outgoing"
2. phone after "partner_id" w/ "invisible when picking_type_code != outgoing"
3.patrner_id after "partner_id" w/ "invisible when picking_type_code != outgoing" 
This sequence give us original layout in outgoing transfer form view.

Reply:
I have checked the layout in both outgoing transfer and incoming transfer.